compiler/sc1.cpp

```diff
@@ -1088,14 +1088,16 @@ static void setconfig(char *root)
 
       /* same for the codepage root */
 # if !defined NO_CODEPAGE
-        *ptr='\0';
+        if(ptr)

```

Comment:
I don't see a way ptr can be NULL here, as it's NULL checked above:

`  if ((ptr=strrchr(path,DIRSEP_CHAR))!=NULL || (ptr=strchr(path,':'))!=NULL) {`

Reply:
i think it’s reassigned and catches the plague here
https://github.com/alliedmodders/sourcepawn/blob/425c86a8f9d40c8bffec7b4bd2e6d32233c66f8f/compiler/sc1.cpp#L1077

Reply:
for reference, here's my trace from the segfault
<img width="588" alt="screen shot 2018-09-28 at 7 16 52 pm" src="https://user-images.githubusercontent.com/11095737/46239828-27aab180-c353-11e8-8b07-69c0f2f5f1b0.png">